src/PerfView/PerfViewData.cs

```diff
@@ -5744,6 +5749,39 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
                                     }
                                     sumCumMetric += cumMetric;
                                     cumCount++;
+
+                                    string GetAllocationType(CallStackIndex csi)
+                                    {
+                                        if (!allocationTypeNames.TryGetValue(csi, out var typeName))
+                                        {
+                                            for (var current = csi; current != CallStackIndex.Invalid; current = eventLog.CallStacks.Caller(current))
```

Comment:
Is there a reasonable bound we can put on number of frames of the stack crawl?  (e.g. are the number of frames below the alloc allways less than say 10?)   That would be nice as it cuts down search by potentially a large number (stacks are often 50 or more, and we are searching every unique stack).

Reply:
Yes, but unfortunately I think 10 is a bit too small. From my limited testing, 15 frames was the longest chain I saw, so I'm thinking of starting with a limit of 20.

Name
 wchar_t
+ wow64!Wow64SystemServiceEx
 + wow64cpu!ServiceNoTurbo
  + wow64cpu!BTCpuSimulate
   + wow64!RunCpuSimulation
    + wow64!Wow64LdrpInitialize
     + ntdll!LdrpInitializeProcess
      + ntdll!??_LdrpInitialize
       + ntdll!LdrpInitialize
        + ntdll!LdrInitializeThunk
         + ntdll!NtTraceEvent
          + ntdll!RtlpLogHeapAllocateEvent
           + ntdll!??RtlpAllocateHeapInternal
            + ntdll!RtlAllocateHeap
             + combase!utInitModuleNameCache

Reply:
20 is fine, indeed 25 is fine (that still avoids 1/2 the cost for almost nothing).